src/pages/task/actionButtons/previousCommits/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { getTaskRoute } from "constants/routes";
+import { LastMainlineCommitQuery } from "gql/generated/types";
+import { reportError } from "utils/errorReporting";
+import { CommitTask, CommitType } from "./types";
+
+// a link cannot be null, so it's common to use # as a substitute.
+const nullLink = "#";
+
+export const getLinks = ({
+  lastExecutedTask,
+  lastPassingTask,
+  parentTask,
+}: {
+  lastExecutedTask: CommitTask;
+  lastPassingTask: CommitTask;
+  parentTask: CommitTask;
+}) => {
+  if (!parentTask) {
+    return {
+      [CommitType.Base]: nullLink,
+      [CommitType.LastPassing]: nullLink,
+      [CommitType.LastExecuted]: nullLink,
+    };
+  }
+
+  return {
+    [CommitType.Base]: getTaskRoute(parentTask.id),
+    [CommitType.LastPassing]: getTaskRoute(
+      lastPassingTask?.id || parentTask.id
+    ),
+    [CommitType.LastExecuted]: getTaskRoute(
+      lastExecutedTask?.id || parentTask.id
+    ),
+  };
+};
+
+// The return value from GetLastMainlineCommitQuery has a lot of nested fields that may or may
+// not exist. The logic to extract the task from it is written in this function.
+export const getTaskFromMainlineCommitsQuery = (
+  data: LastMainlineCommitQuery
+): CommitTask => {
+  const buildVariants =
+    data?.mainlineCommits.versions.find(({ version }) => version)?.version
+      .buildVariants ?? [];
+  if (buildVariants.length > 1) {
+    reportError(
+      new Error("Multiple build variants matched previous commit search.")
```

Comment:
Is there more info here that we can add to the sentry search? Or will sentry give us the necessary data be default?

Reply:
Sentry will include a breadcrumb with the GraphQL query sent before this error. I think that's enough information to start investigating this.